docs/core/compatibility/serialization/8.0/publishtrimmed.md

```diff
@@ -0,0 +1,51 @@
+---
+title: "Breaking change: PublishedTrimmed projects fail reflection-based serialization"
+description: Learn about the .NET 8 breaking change in System.Text.Json serialization where projects with PublishTrimmed enabled now fail reflection-based serialization by default.
+ms.date: 11/08/2023
+---
+# PublishedTrimmed projects fail reflection-based serialization
+
+Projects that enable the [PublishTrimmed](../../../deploying/trimming/trimming-options.md#enable-trimming) MSBuild property now automatically turn off the reflection-based defaults of [System.Text.Json](../../../../standard/serialization/system-text-json/overview.md). In other words, setting `PublishTrimmed` to `true` automatically sets the `JsonSerializerIsReflectionEnabledByDefault` MSBuild property to `false` unless otherwise specified in project configuration.
+
+## Previous behavior
+
+Prior to this change, projects that have the `PublishTrimmed` property enabled, that is, `<PublishTrimmed>true</PublishTrimmed>`, published a trimmed application. However, the reflection-based default serialization behavior wasn't necessarily disabled. Depending on what code got trimmed, the follow code might or might not succeed serialization, or might or might not output the correct serialization data.
+
+```csharp
+JSonSerializer.Serialize(new { Value = 42 });
+```
+
+## New behavior
+
+Starting in .NET 8, projects that have the `PublishTrimmed` property enabled fail serialization outright. The code `JSonSerializer.Serialize(new { Value = 42 });` throws the following exception:
+
+> **System.InvalidOperationException: Reflection-based serialization has been disabled for this application.**
+
+## Version introduced
+
+.NET 8 Preview 7
```

Comment:
I put this on the breaking change issue, but I wonder now if specifying the preview version is relevant for documentation purposes.

Reply:
It's really only relevant before GA. I've removed the "introduced version" column from the breaking changes overview page for .NET 8, but I don't think it hurts to leave it here.